README.md

```diff
@@ -1,22 +1,164 @@
+our `tsconfig.json` file.
+
+## The tsconfig.json file
+
+So what _is_ the tsconfig.json file? The tsconfig file is used to help us
+configure and compile our code! This file is _always_ kept in the root of our
+project, and is used to identify the compiler rules we want to enforce, as well
+as specify the root files of the project. The rules the compiler enforces are
+completely customizable and up to you to determine based on your project needs.
+
+There's unlimited ways that you can setup your tsconfig.json file, and a lot of
+different properties you can add to it.
+
+Let's create our tsconfig.json file in the root of our project directory by
+running:
+
+```zsh
+$ touch tsconfig.json
+```
+
+Awesome. Now let's fill in our properties. Were going to start with a
+`"compilerOptions"` key with an object as it's value. The `"compilerOptions"`
+object property is what we will use to declare the rules we want to enforce in
+our project. Inside of the `"compilerOptions"` object, we're going to add 3
+properties.
+
+Let's start by adding the property`"module"`, and setting it's value to
+`"commonjs"`. This tells TypeScript that we intend to use `commonjs` syntax for
+importing and exporting modules. Next, add the `"target"` property.
+
+The `"target"` property is used to tell TypeScript what version of ECMAscript we
+will be using. Because all modern browsers are able to use `es6`, let's set that
+value to `"es6"`. If we were sure we would only be using newer or updated
+browsers we could set our value to include the year (e.g. `es2022`), or even
+`esnext`.
+
+Our final property to add is`"strictNullChecks"`, which makes sure no variable
+has the value of `null` or `undefined` unless we explicitly assign them.
+
+Let's add one more property to our json object, `"include": ["**/*.ts"]`. The
+`include` property tells our compiler what files to check our rules against. Our
+value, `["**/*.ts"]`, tells the compiler to run these rules against every file
+in our project that has the extension `.ts`.
+
+You should now have an object that looks something like this:
+
+```json
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "target": "es6",
+    "strictNullChecks": true
+  },
+  "include": ["**/*.ts"]
+}
+```
+
+For this exercise, we're working with a very simple `tsconfig.json` file, but
+it's definitely worth taking a look at the [TypeScript Docs - tsconfig][] for
+more compiler options and properties to add to your projects in the future!
+
+## Compiling with `tsc`
+
+- run tsc to see the errors
+- then fix the errors
+- then run tsc to see there are no more errors
+- run node index.js to run our code
+- submit code
+
+Now that we have our `tsconfig.json` setup, we're able to compile our code using
+the `tsc` command! There's a ton of different compiler options for us (check out
+[TypeScript Docs - Compiler Options][] to see them all!) to use... we can use
+the command by adding a specific file to it like `npx tsc index.ts`, or we can
```

Comment:
Suggest moving the parenthetical info after "to use"

Reply:
```suggestion
Now that we have our `tsconfig.json` setup, we're able to compile our code using
the `tsc` command! There's a ton of different compiler options for us to use  (check out
[TypeScript Docs - Compiler Options][] to see them all!), we can use
the command by adding a specific file to it like `npx tsc index.ts`, or we can
```